src/lib/stores/walletStore.js

```diff
@@ -85,28 +85,67 @@ function createWalletStore() {
-         * Sign and return a Stellar transaction
+         * Sign and return a Stellar transaction, handling both regular payments and SAC transfers
          * @param {Object} opts Options object
-         * @param {string} opts.transactionXDR A Stellar transaction in base64-encoded XDR format
+         * @param {string} opts.destination Destination address (account or contract)
+         * @param {string} opts.amount Amount to send
+         * @param {Asset} opts.asset Asset to send
          * @param {string} opts.network Network passphrase for the network this transaction is intended for
          * @param {string} opts.pincode Pincode to be used as the encryption password for the keypair
-         * @returns {Promise<Transaction>} A signed Stellar transaction ready to submit to the network
+         * @returns {Promise<import('stellar-sdk').Transaction>} A signed Stellar transaction ready to submit to the network
          * @throws Will throw an error if there is a problem signing the transaction.
          */
-        sign: async ({ transactionXDR, network, pincode }) => {
+        signPayment: async ({ destination, amount, asset, network, pincode }) => {
             try {
                 const keyManager = setupKeyManager()
+                const { publicKey } = get(walletStore)
+                const sourceAccount = new Account(publicKey, '0') // you have to Replace '0' with actual sequence number
+
+                let transaction;
+
+                if (contacts.isContractAddress(destination)) {
+                    // Handle SAC transfer
+                    const contract = new Contract(destination);
+                    const transferOp = contract.call(
+                        'transfer',
+                        xdr.ScVal.scvAddress(Address.fromString(publicKey).toScAddress()),
+                        xdr.ScVal.scvAddress(Address.fromString(destination).toScAddress()),
+                        xdr.ScVal.scvI128(new xdr.Int128Parts({
+                            lo: xdr.Uint64.fromString(amount),
+                            hi: xdr.Int64.fromString('0')
+                        })),
+                        xdr.ScVal.scvSymbol(asset.getCode())
+                    );
+
+                    transaction = new TransactionBuilder(sourceAccount, { fee: '100', networkPassphrase: network })
+                        .addOperation(transferOp)
+                        .setTimeout(30)
+                        .build();
+                } else {
+                    // Handle regular payment
+                    transaction = new TransactionBuilder(sourceAccount, { fee: '100', networkPassphrase: network })
+                        .addOperation(Operation.payment({
+                            destination: destination,
+                            asset: asset,
+                            amount: amount
+                        }))
+                        .setTimeout(30)
+                        .build();
+                }
+
```

Comment:
I _think_ these changes are leftover from the previous PR? If so, let's remove them here. Philosophically, we **really want** this function to take a transaction that the user has _already approved_, and only attach a signature. We don't want to build the transaction after they've approved "something." The tx building (and yes, simulating SAC transfers) need to be done _prior_ to them pushing the `sign` button, so they know exactly what they're signing.